java/java.source.base/src/org/netbeans/modules/java/source/indexing/JavaCustomIndexer.java

```diff
@@ -147,6 +151,76 @@ protected void index(final Iterable<? extends Indexable> files, final Context co
                 JavaIndex.LOG.fine("Ignoring request with no root"); //NOI18N
                 return;
             }
+            BinaryForSourceQuery.Result2 binRes = BinaryForSourceQuery.findBinaryRoots2(root.toURL());
```

Comment:
There is an additional call to project query but should not be a big problem as it's followed by query for bootpath, compilepath, sourcepath, source level and encoding. Also it's O(n) to roots.